vcd/resource_vcd_rde_type.go

```diff
@@ -0,0 +1,337 @@
+package vcd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+)
+
+func resourceVcdRdeType() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdRdeTypeCreate,
+		ReadContext:   resourceVcdRdeTypeRead,
+		UpdateContext: resourceVcdRdeTypeUpdate,
+		DeleteContext: resourceVcdRdeTypeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdRdeTypeImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the Runtime Defined Entity Type",
+			},
+			"vendor": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The vendor name for the Runtime Defined Entity Type",
+			},
+			"nss": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A unique namespace associated with the Runtime Defined Entity Type",
```

Comment:
Same thought as above: maybe we should add in the three Descriptions above that these three properties compose a single unique key?

Reply:
Done, same in `rde_interface`.